docs/writing_review_screen.md

```diff
@@ -0,0 +1,240 @@
+3. Remove edit questions about kickout screens. Sometimes, you will have questions that do set a variable in the interview, but it's only use is to kickout a user if their problem isn't covered by the form. Here's an example of one:
+
+```yml
+---
+id: Is this a probate case
+question: |
+  Are you Filing in a Probate Case?
+subquestion: |
+  Is the case a probate case? Examples of probate cases include cases to distribute property after someone dies, guardianships, and conservatorships. 
+fields:
+  - Is your case a probate case?: probate_case
+    datatype: yesnoradio
+---
+event: probate_case_exit
+question: |
+  You cannot continue to use the tool.
+subquestion: |
+  Based on your answer, you can't use this tool to make a Notice of Hearing and Motion. Probate cases have different procedural rules than other cases. You may want to speak to a lawyer.
+  
+  If you made a mistake, click the **Undo** button. Otherwise, click the **Exit** button to return to Michigan Legal Help.
+buttons:
+  - Exit: exit
+    url: https://michiganlegalhelp.org
+```
+
+If editing the value of a variable like this would only lead the person to a kickout screen and not change any other logic of the interview, then the generated review section should be removed.
+
+
+### Step 4: Things to add to the review screen
+
+1. You'll find that certain variables often depend on each other, either explicitly or implicitly. For example, if you have a variable called `hearing_date_and_time` that you set to `hearing_date + " " + hearing_time`, you should recompute that field whenever the user edits the hearing date or time. You can do something like this to update the variable whenever you edit the hearing date:
+
+```yml
+  - label: Edit
+    fields:
+      - hearing_date
+      - recompute:
+        - hearing_date_and_time
+    button: |
+      ...
+```
+
+Another example of when two variables are implicitly dependent would be if you asked the user if the plaintiff or defendant is paying child support. If the user changes what side of the case they are on, their answer to what side is paying might change too! You can use a review entry that looks like this to edit both.
+
+```yml
+  - label: Edit
+    fields:
+      - user_ask_role
+      - who_paid
+```
+
+2. As mentioned in step 2, if a variable doesn't exist or any errors occur when showing a section of the review screen, that whole section won't be shown. If some of the fields on a screen or optional, then their absence can cause their section of the review screen to not appear. Use `showifdef` on optional fields, or fields that use `show if`, to prevent this issue.
+3. While writing the review screen, you will likely have to redo your interview logic, so that it doesn't make assumptions about the presence or absence of variables, just their values, which I'd encourage. From the docassemble documentation:
+
```

Comment:
This is a great point. I think it needs its own heading. Maybe:

Defensive interview writing: Writing your interview so that answers can be changed by the user